Mage.Sets/src/mage/cards/d/DemonOfFatesDesign.java

```diff
@@ -0,0 +1,261 @@
+
+    private DemonOfFatesDesignPayLifeCost(final DemonOfFatesDesignPayLifeCost cost) {
+        super(cost);
+        this.mor = cost.mor;
+    }
+
+    @Override
+    public DemonOfFatesDesignPayLifeCost copy() {
+        return new DemonOfFatesDesignPayLifeCost(this);
+    }
+
+    MageObjectReference getMor() {
+        return this.mor;
+    }
+
+    void setMor(MageObjectReference mor) {
+        this.mor = mor;
+    }
+
+}
+
+enum DemonOfFatesDesignCost implements DynamicCost {
+    instance;
+
+    @Override
+    public Cost getCost(Ability ability, Game game) {
+        return new DemonOfFatesDesignPayLifeCost(ability.getManaCosts().manaValue());
+    }
+
+    @Override
+    public String getText(Ability ability, Game game) {
+        return "Pay " + ability.getManaCosts().manaValue() + " life rather than "
+                + ability.getManaCosts().getText() + '?';
+    }
+}
+
+class DemonOfFatesDesignAlternativeCostSourceAbility extends AlternativeCostSourceAbility {
+
+    private MageObjectReference mor;
+
+    DemonOfFatesDesignAlternativeCostSourceAbility() {
+        super(
+                new CompoundCondition(SourceIsSpellCondition.instance, IsBeingCastFromHandCondition.instance),
+                null, StaticFilters.FILTER_CARD_ENCHANTMENT,
+                true, DemonOfFatesDesignCost.instance
+        );
+    }
+
+    private DemonOfFatesDesignAlternativeCostSourceAbility(final DemonOfFatesDesignAlternativeCostSourceAbility effect) {
+        super(effect);
+    }
+
+    @Override
+    public DemonOfFatesDesignAlternativeCostSourceAbility copy() {
+        return new DemonOfFatesDesignAlternativeCostSourceAbility(this);
+    }
+
+    @Override
+    protected void doActivate(Game game, Ability ability) {
+        super.doActivate(game, ability);
+        for (Cost cost : ability.getCosts()) {
+            if (cost != null && cost instanceof DemonOfFatesDesignPayLifeCost) {
+                ((DemonOfFatesDesignPayLifeCost) cost).setMor(getMor(game));
+            }
+        }
+    }
+
+    MageObjectReference getMor(Game game) {
+        return new MageObjectReference(getSourceObject(game), game);
+    }
+}
+
+class DemonOfFatesDesignCastEffect extends ContinuousEffectImpl {
+
+    private final DemonOfFatesDesignAlternativeCostSourceAbility alternativeCastingCostAbility
+            = new DemonOfFatesDesignAlternativeCostSourceAbility();
+
+    DemonOfFatesDesignCastEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral);
+        this.staticText = "You may cast an enchantment spell by paying life equal to its mana value rather than paying its mana cost.";
+    }
+
+    private DemonOfFatesDesignCastEffect(final DemonOfFatesDesignCastEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public DemonOfFatesDesignCastEffect copy() {
+        return new DemonOfFatesDesignCastEffect(this);
+    }
+
+    @Override
+    public boolean apply(Layer layer, SubLayer sublayer, Ability source, Game game) {
```

Comment:
Why use the four-argument method here rather than the two-argument method?

Reply:
I am not sure actually, the different similar alternative casting effect all have init & 4-argument apply with code, and 2-argument apply returning false.

Reply:
Using 2-argument apply does not work apparently, so I'll keep the 4-argument one:
![image](https://github.com/magefree/mage/assets/34709007/84c8bb73-8270-4cb9-9dba-cb47d02b3276)

Reply:
Huh. When you use the two-argument apply(), then you have to use the super() constructor with the layer/sublayer arguments. If you did that and it still doesn't work, then fine as is. I wonder why.

Reply:
Oh right, let me try it that way.

Reply:
Ok it does work with the proper constructor (since the 4-argument method from the ContinuousEffect's class is then calling the 2-argument one from the Demon).

This PR is quite big already, so I will not clean the other similar cards. Can be a followup task though.